extensions/notebook/src/jupyter/jupyterNotebookProvider.ts

```diff
@@ -46,15 +51,52 @@ export class JupyterNotebookProvider implements nb.NotebookProvider {
 			// As this is a notification method, will skip throwing an error here
 			return;
 		}
-		let uriString = notebookUri.toString();
-		let manager = this.managerTracker.get(uriString);
+		let baseFolder = this.transformToBaseFolder(notebookUri.fsPath.toString());
+		let manager = this.managerTracker.get(baseFolder);
 		if (manager) {
-			this.managerTracker.delete(uriString);
-			manager.dispose();
+			let sessionManager = (manager.sessionManager as JupyterSessionManager);
+			let session = sessionManager.listRunning().find(e => e.path === notebookUri.fsPath);
```

Comment:
It looks like this cast is only necessary because of the listRunning call - what about adding that to the SessionManager interface? That seems to be a good thing to have anyways